src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -576,14 +576,17 @@ public ProjectState WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyz
                 projectInfo: this.ProjectInfo.WithAnalyzerReferences(analyzerReferences).WithVersion(this.Version.GetNewerVersion()));
         }
 
-        public ProjectState AddDocument(DocumentState document)
+        public ProjectState AddDocuments(ImmutableArray<DocumentState> documents)
         {
-            Debug.Assert(!this.DocumentStates.ContainsKey(document.Id));
+            foreach (var document in documents)
```

Comment:
`Debug.Assert(!documents.Any(document => ...))` to avoid the foreach being emitted in release mode.